From ebc92bb115137bcbf2766c8edbea8710c2075b95 Mon Sep 17 00:00:00 2001
From: Harsha Shanthaveeraiah <harshax.shanthaveeraiah@intel.com>
Date: Wed, 8 Aug 2012 14:41:05 +0530
Subject: [PORT FROM MAIN] WORK ARROUND::Audio Effects: MTBF ACS test of "Effect Preset Reverb during audio playback IHF" TOMEBSTONE occur

BZ: 45836

- This patch has the changes to double the MAX buffer size allocated to
  the Input buffer of the reverb engine which is used for Aux Reverb.
- LVREV_MAX_FRAME_SIZE value is doubled in Reverb engine which is local to the
  reverb context.
- This is to avoid the crash happening while freeing the Aux buffer in Audioflinger.
  Since the buffer size was doubled due to the HAL changes, LVREV_MAX_FRAME_SIZE
  was also doubled to avoid the overflow while copying the input buffer.

  DO NOT UPSTREAM

Change-Id: If9e47afc20c6edf7150eeade750099570b56f087
Orig-Change-Id: Ifa5ab6a03e34c3fdf45bb46ecda83727397b18c1
Reviewed-on: http://android.intel.com:8080/82105
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libeffects/lvm/wrapper/Reverb/EffectReverb.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libeffects/lvm/wrapper/Reverb/EffectReverb.h b/media/libeffects/lvm/wrapper/Reverb/EffectReverb.h
index 7c15b18..d47fe04 100644
--- a/media/libeffects/lvm/wrapper/Reverb/EffectReverb.h
+++ b/media/libeffects/lvm/wrapper/Reverb/EffectReverb.h
@@ -28,7 +28,7 @@ extern "C" {
 #define MAX_CALL_SIZE           256
 #define LVREV_MAX_T60           7000
 #define LVREV_MAX_REVERB_LEVEL  2000
-#define LVREV_MAX_FRAME_SIZE    2560
+#define LVREV_MAX_FRAME_SIZE    2560*2 //Max frame size is doubled to support increase in framecount from HAL
 #define LVREV_CUP_LOAD_ARM9E    470    // Expressed in 0.1 MIPS
 #define LVREV_MEM_USAGE         71+(LVREV_MAX_FRAME_SIZE>>7)     // Expressed in kB
 //#define LVM_PCM
-- 
1.7.4.1

